src/fiatconnect/ReviewScreen.tsx

```diff
@@ -61,24 +96,20 @@ export default function FiatConnectReviewScreen({ route, navigation }: Props) {
     [normalizedQuote, defaultLocaleCurrencyCode]
   )
 
-  const feeType = FeeType.SEND
-  const tokenList: TokenBalance[] = useSelector(tokensListWithAddressSelector)
-  const cryptoType = normalizedQuote.getCryptoType()
-  const tokenAddress = tokenList.find((token) => token.symbol === cryptoType)?.address
-  const feeEstimates = useSelector(feeEstimatesSelector)
-  const feeEstimate = tokenAddress ? feeEstimates[tokenAddress]?.[feeType] : undefined
-  const usdTokenInfo = useTokenInfo(networkConfig.cusdTokenId)!
-  const networkFee =
-    useLocalToTokenAmount(
-      feeEstimate?.usdFee ? new BigNumber(feeEstimate?.usdFee) : new BigNumber(0),
-      usdTokenInfo.tokenId
-    ) ?? new BigNumber(0)
+  const tokenId = normalizedQuote.getTokenId()
+  const token = useTokenInfo(tokenId)
 
-  useEffect(() => {
-    if (!feeEstimate && tokenAddress) {
-      dispatch(estimateFee({ feeType, tokenAddress }))
-    }
-  }, [feeEstimate, tokenAddress])
+  const transferResult = useAsync(async () => {
```

Comment:
yeah i think these differences are mainly down to the "old" way that we send transactions (guestimate and send and monitor) vs using the prepared transactions (do it upfront, don't send transactions that can't succeed). if we don't prepare the transaction upfront, we can't show the combined fees including the network fee. and to prepare the transaction, we need to fetch the provider address.

is it an issue to hit the provider endpoint to fetch their address more often? (e.g. does it mess with their ability to handle the traffic or analytics etc.) if so, i think we can prepare transactions using a dummy address and update it on submit.

Reply:
Yeah this all makes sense to me - I guess my (nitpicky) gripe is that the semantics are kinda wonky here. The `transfer` endpoint for FC providers is actually a `POST` which (presumably) creates a new ongoing transfer somewhere in the provider's DB associated with the given quote. It's expected that sending this request corresponds with a user's intent to follow through with the quote (it also "locks up" the quoteId in question so it can't be used for another transfer), and doing  that without the user pressing "confirm transfer" or whatever feels kinda wrong to me.

Realistically, it doesn't _really_ matter, which is why I say this is all a nitpick. Nothing will (should) break on the provider's end if we initiate these transfers but don't follow through with them (the user never presses "confirm").

One potential UX side effect on our end of doing things this way might be the following flow:
* User goes to review screen for a quote
* We hit the `transfer` endpoint, locking up the `quoteId`
* User goes back to select provider screen
* User returns to the same review screen
* We hit the `transfer` endpoint again with the same `quoteId`, getting a 409 this time
* The user can never actually complete the quote they got with that provider; they have to fetch totally new quotes for all providers.

I'm pretty sure this is what would happen given how our fetching works, but not totally positive.

Reply:
Ahh yeah, gotcha - yeah my vote's definitely for this approach as well. i don't think there should be any harm in it, since the gas estimation ought to be agnostic to the target address. i don

> as long as the provider has some non zero balance of the cash out token

why does this need to be true in order for the gas estimation to be the same whether or not we use the user's address/provider's address as the beneficiary in the "dummy" transaction?

Reply:
i was chatting about this with Jean yesterday and in theory, updating an existing entry in a map is less work than creating a new entry in a map - presumably, the user has a balance of the token they're trying to cash out so we estimate updating an existing balance, but if the provider does not have a balance of that token then a new entry needs to be created on that contract. i think this is an unlikely scenario so i didn't think much more on it or build a solution for it (i kind of expect that the difference would be negligable, and a user would only encounter this issue if trying to send really close to their max balance too)